### PR TITLE
Small fixes in examples / simple-interfaces

### PR DESCRIPTION
**Pull request by andreasthoelke**
This example was very helpful to me, thank you.

I noticed some small inconsistencies.

### CODE REVIEW

**Comment by andreasthoelke** [comment]
Btw, happy to apply the same to the `simple-classes` example (`examples/simple-classes/src/schema.ts`) if I got this right.

**Comment by hayes** [comment]
> Btw, happy to apply the same to the `simple-classes` example (`examples/simple-classes/src/schema.ts`) if I got this right.

That would be great!

File: examples/simple-interfaces/src/schema.ts [hunk 1]
```diff
@@ -48,11 +48,11 @@ Comment.implement({
...
       type: User,
       nullable: true,
-      resolve: (comment) => Users.get(comment.id),
+      resolve: (comment) => [...Users.values()].find((user) => user.id === comment.authorId),
     }),
     post: t.field({
       type: Post,
-      resolve: (comment) => Posts.get(comment.id)!,
+      resolve: (comment) => [...Posts.values()].find((post) => post.id === comment.postId),
     }),
```
Comment (hayes) [inline examples/simple-interfaces/src/schema.ts:56]: I think this needs a non-null assertion, or the field needs to be made nullable, looks like there is a type error when running the typescript on this example

Comment (andreasthoelke) [reply examples/simple-interfaces/src/schema.ts:56]: Right. It's great that typescript/pothos catches this. I had only tested in a different project before where the typescript error did not show, it seems I'm missing a related tsconfig setting.

I think, in the case of the `Comment` type it makes sense to assume that the database schema assures that a comment can only exist with a reference to a post. Whereas a comment might have no (/an anonymous) author. I propose that we thus assert that the `post` field’s `resolve` value is non-null (using the `!` operator) whereas the `author` field is `nullable: true`.

`pnpm run ci` now succeeds.

I'll still add `examples/simple-classes` to this pr.

**Comment by hayes** [comment]
Thanks so much for these fixes!

**Pull request merged**